src/compas_timber/_fabrication/dovetail_mortise.py

```diff
@@ -0,0 +1,727 @@
+
+        if isinstance(frame, Plane):
+            frame = Frame.from_plane(frame)
+
+        # define ref_side & ref_edge
+        ref_side = beam.ref_sides[ref_side_index]
+        ref_edge = Line.from_point_and_vector(ref_side.point, ref_side.xaxis)
+
+        # calculate orientation
+        orientation = cls._calculate_orientation(ref_side, frame)
+
+        # calclulate start_x and start_y
+        start_x = cls._calculate_start_x(ref_side, ref_edge, frame)
+        start_y = abs(frame.point[2] - ref_side.point[2])
+
+        # define angle
+        if orientation == OrientationType.START:
+            angle -= 90.0
+        else:
+            angle += 90.0
+
+        # define slope and inclination
+        # TODO: In which cases do you want indiferent slope and inclination?
+        slope = 90.0
+        inclination = 90.0
+
+        # determine if the top and bottom length of the cut is limited
+        limitation_top = LimitationTopType.UNLIMITED
+        length_limited_bottom = True
+
+        use_flank_angle = True if flank_angle != 15.0 else False  # TODO: does this change anything?
+
+        return cls(
+            start_x,
+            start_y,
+            start_depth,
+            angle,
+            slope,
+            inclination,
+            limitation_top,
+            length_limited_bottom,
+            length,
+            width,
+            depth,
+            cone_angle,
+            use_flank_angle,
+            flank_angle,
+            shape,
+            shape_radius,
+            ref_side_index=ref_side_index,
+        )
+
+    @staticmethod
+    def _calculate_orientation(ref_side, cutting_frame):
+        # calculate the orientation of the beam by comparing the xaxis's direction of the ref_side and the plane.
+        # Orientation is not set as a param for the BTLxDovetailMortise processing but its essential for the definition of the rest of the params.
+        perp_plane = Plane(cutting_frame.point, cutting_frame.xaxis)
+        if is_point_behind_plane(ref_side.point, perp_plane):
+            return OrientationType.END
+        else:
+            return OrientationType.START
+
+    @staticmethod
+    def _calculate_start_x(ref_side, ref_edge, cutting_frame):
+        # calculate the start_x of the cut based on the ref_side, ref_edge and cutting_frame
+        perp_plane = Plane(cutting_frame.point, ref_side.xaxis)
+        point_start_x = intersection_line_plane(ref_edge, perp_plane)
+        if point_start_x is None:
+            raise ValueError("Plane does not intersect with beam.")
+
+        start_x = distance_point_point(ref_side.point, point_start_x)
+        return start_x
+
+    ########################################################################
+    # Class Methods
+    ########################################################################
+
+    @classmethod
+    def define_dovetail_tool(self, tool_angle, tool_diameter, tool_height):
```

Comment:
in python we name the first argument of a class method `cls` instead of `self`

Reply:
missed that one too. thanks